src/test/compile-fail/const-size_of-cycle.rs

```diff
@@ -8,11 +8,12 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// error-pattern: unsupported cyclic reference between types/traits detected
```

Comment:
Change it into a ui test?

Reply:
Oh, I just realized this is a bit suboptimal. Or rather, the span is lost of some reason.

Reply:
Oh this is a bug with `ParamEnvAnd` `Key` impl for the query system!

Reply:
Welp, no, this is the best I get right now:
```rust
error[E0391]: unsupported cyclic reference between types/traits detected
   --> /home/eddy/Projects/rust-2/src/libcore/mem.rs:192:14
    |
192 |     unsafe { intrinsics::size_of::<T>() }
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^ cyclic reference
    |
```
There is a note at the bottom of the cycle which shows the expression that triggered the cycle, so not all is lost!

However, I believe there are two `ParamEnv`'s with *different* `Reveal` modes on the cycle stack, so the cycle isn't detected early enough... cc @nikomatsakis